src/plugins/service-worker/index.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+const buildPaths = require('mwp-cli/src/commands/buildCommands/config/paths');
+const path = require('path');
+
+/*
+ * The service worker plugin will register a root-level route for serving the
+ * service worker script file.
+ *
+ * Disk location of the script file is determined by builder CLI config import.
+ */
+export default function register(
+	server: Object,
+	options: ?Object,
+	next: () => void
+) {
+	/*
+	 * Route for service worker script at top-level path. Depends on `Inert`
+	 * `path` must match client `serviceWorker.register` call - MWP provides this
+	 * in the `<ServiceWorker>` component
+	 */
+	server.route({
+		method: 'GET',
+		path: '/asset-service-worker.{localeCode}.js',
```

Comment:
The goal of the ticket seems to have migrated a bit (which is fine) and its taking me a moment to catch on. 

My initial expectation of service worker implementation was the basic caching the browser already does, but functional without a network connection. I thiiink thats what this does, in part - https://github.com/GoogleChrome/samples/blob/gh-pages/service-worker/selective-caching/service-worker.js

Anyway, thats what I mean by 'caching behavior is implementation specific' - its not a service worker thing, but how you implement the service worker.

Precaching - a term I was misunderstanding - means front loading assets. I suspect we don't want to front load all assets for our site, at least not in the long term. Changing asset loading like this is worthy of group discussion.

Finally, I'm confused about how the service worker cache is cleared.

---

While there are items to discuss, my primary concern is "will this confuse developers?" if the answer is no, then carry on and we can decide later whether or not to change course.